CONTRIBUTING.md

```diff
@@ -8,6 +8,14 @@ This project adheres to the [Contributor Covenant Code of Conduct](http://contri
 
 The majority of contributions won't need to touch any Ruby code at all.
 
+### Dependencies
+
+Linguist uses the [`charlock_holmes`](https://github.com/brianmario/charlock_holmes) character encoding detection library which in turn uses [ICU](http://site.icu-project.org/), and the libgit2 bindings for Ruby provided by [`rugged`](https://github.com/libgit2/rugged). [Bundler](https://bundler.io/) v1.10.0 or newer is required for installing the Ruby gem dependencies. [Docker](https://www.docker.com/) is also required when adding or updating grammars. These components have their own dependencies - `icu4c`, and `cmake` and `pkg-config` respectively - which you may need to install before you can install Linguist.
+
+For example, on macOS with [Homebrew](http://brew.sh/): `brew install cmake pkg-config icu4c && brew cask install docker` and on Ubuntu: `apt-get install cmake pkg-config libicu-dev docker-ce`.
+
+The latest version of Bundler v1 can be installed with `gem install bundler -v "~>1.10"`.
+
```

Comment:
I've moved the dependencies above "Getting started" as it's a lot easier to get started once you've met all the dependencies 😁